src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [ExportLspMethod(Methods.TextDocumentCompletionName)]
+    internal class CompletionHandler : RazorLSPHandlerBase, IRequestHandler<CompletionParams, SumType<CompletionItem[], CompletionList>?>
+    {
+        private static readonly string[] CSharpTriggerCharacters = new[] { ".", "@" };
+        private static readonly string[] HtmlTriggerCharacters = new[] { "<", "&", "\\", "/", "'", "\"", "=", ":" };
```

Comment:
I copied this from the HTML language server. @ToddGrun, is there a way I can get this without copying this over?

Reply:
These come back as part of the initializion results, not sure if the are exposed outside of that.